contracts/strategy/StrategyBaseV2.sol

```diff
@@ -200,11 +214,37 @@ abstract contract StrategyBaseV2 is IStrategyV2, ControllableV3 {
 
     uint amountAdjusted = Math.min(amount, balance);
     if (amountAdjusted != 0) {
+      _decreaseBaseAmount(_asset, amountAdjusted);
       IERC20(_asset).safeTransfer(_splitter, amountAdjusted);
     }
     emit WithdrawToSplitter(amount, amountAdjusted, balance);
   }
 
+
+  // *************************************************************
+  //                  baseAmounts modifications
+  // *************************************************************
+
+  /// @notice Decrease {baseAmounts} of the {asset} on {amount_}
+  ///         The {amount_} can be greater then total base amount value because it can includes rewards.
+  ///         We assume here, that base amounts are spent first, then rewards and any other profit-amounts
+  function _decreaseBaseAmount(address asset_, uint amount_) internal {
+    uint baseAmount = baseAmounts[asset_];
+    baseAmount = baseAmount > amount_
+      ? baseAmount - amount_
+      : 0;
```

Comment:
why not revert?

Reply:
_decreaseBaseAmount doesn't revert because we need possibility to call this function in inheritors. And inheritors will use  this function to register/unregister all available amoutns. F.e. in ConverterStrategyBase._depositToPool() deposits all available amount (both base amount and rewards). After that it should call _decreaseBaseAmount to decrease base-amount to zero.  Imho it worth _depositToPool() is not good place to check what part of balance is base-amount and what is rewards - better to do it inside _decreaseBaseAmount.

Reply:
Sorry, you are right. I will change it to revert. Rewards should be divided on two parts: the part to forwarder and the part for strategy. And the part for strategy should increase base amount.